utils/load_image.py

```diff
@@ -0,0 +1,90 @@
+#%%
+import tarfile
+from io import BytesIO
+from PIL import Image
+import os
+from pathlib import Path
+import pandas as pd
+
+
+basefolder_loc = Path(__file__).parents[1]
+TARFILE = tarfile.open(
+    os.path.join(basefolder_loc, "1.download-data", "data", "training.tar.gz"), "r:gz",
+)
+
+
+def load_img(
+    target: str = "adrenoceptor",
+    plate: str = "P1",
+    cell_id: int = 1,
+    replicate: int = 1,
+    wellName: str = "C10",
+    field: int = 1,
+) -> Image:
+    """
+    Find the single image that matches on all given parameters.
+
+    Returns: image
+
+    WARNING:
+    Use this only if you need to load a single image.
+    Every time you run this function it loops over the whole zip file.
+    If you need all images use get_all_images.
+    """
+    newPlateName = plate.replace("P", "S")
+    extracted_file = TARFILE.extractfile(
+        f"training/{target}/211_11_17_X_Man_LOPAC_X5_LP_{newPlateName}_{replicate}_{wellName}_{field}_{cell_id}.tiff"
+    )
+    b = extracted_file.read()
+
+    img = Image.open(BytesIO(b))
+    return img
+
+
+def get_all_images(dataframe: pd.DataFrame) -> (Image, str, str):
+    """
+    Creates a generator with all the image that match exactly one row in dataframe.
```

Comment:
```suggestion
    Creates a generator with all the images that match exactly one row in the dataframe.
```

still not sure what the "dataframe" is supposed to be. Maybe a more descriptive variable would help?

Also, how does one use this function? Maybe some usage notes would be helpful